bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/details/ConicNameEditor.scala

```diff
@@ -41,6 +41,10 @@ final class ConicNameEditor(date: HorizonsIO[Date]) extends JPanel with Telescop
       override def textBoxKeyPress(tbwe: TextBoxWidget): Unit =
         nonreentrant {
           spt.getTarget.setName(tbwe.getValue)
+          // editing the name so invalidate the horizons id.
```

Comment:
One thing to be a bit careful about is to avoid calling a method with a `nonreentrant` section from within a `nonreentrant` section.  Otherwise when it bubbles back up the editing flag will be off before the end of the outer `nonreentrant` block.